ee/localserver/krypto-ec-middleware.go

```diff
@@ -0,0 +1,178 @@
+package localserver
+
+import (
+	"crypto"
+	"crypto/ecdsa"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"github.com/kolide/krypto/pkg/challenge"
+	"github.com/kolide/krypto/pkg/echelper"
+	"github.com/vmihailenco/msgpack/v5"
+)
+
+type kryptoEcMiddleware struct {
+	signer       crypto.Signer
+	counterParty ecdsa.PublicKey
+	logger       log.Logger
+}
+
+func newKryptoEcMiddleware(logger log.Logger, signer crypto.Signer, counteParty ecdsa.PublicKey) *kryptoEcMiddleware {
+	return &kryptoEcMiddleware{
+		signer:       signer,
+		counterParty: counteParty,
+		logger:       logger,
+	}
+}
+
+func (e *kryptoEcMiddleware) unwrap(next http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if r.Body != nil {
+			r.Body.Close()
+		}
+
+		// Extract the box from the URL query parameters
+		boxRaw := r.URL.Query().Get("box")
+		if boxRaw == "" {
+			level.Debug(e.logger).Log("msg", "no data in box query parameter")
+			w.WriteHeader(http.StatusUnauthorized)
+			return
+		}
+
+		outerChallenge, err := marshalOuterChallenge(boxRaw)
```

Comment:
I haven't absorbed this, but I'm somewhat surprised? I expected this would be using `challenge.Respond`

Reply:
well we have to figure out what the actual command request is, to do that we have to open the box. When we actually get to the wrap function, we can call `challenge.Respond`